[18.0][MIG] stock_quant_history

bosd opened the pull request:
Standard migration using odoo-module-migrator /ocabot migration stock_quant_history .

@petrus-v 
Shall we first migrate this base, then forward port all the improvements, from V14 and V17?
As some of the "improvement" PR's are still open and have some review comments.

petrus-v committed:
[ADD] stock_quantity_history: re-generate stock.quant at

Commit:
[UPD] Update stock_quant_history.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Added translation using Weblate (Italian)

nico-foodles committed:
[IMP] stock_quant_history: pre-commit auto fixes

nico-foodles committed:
[MIG] stock_quant_history: Migration to 17.0

Commit:
[UPD] Update stock_quant_history.pot

OCA-git-bot committed:
[BOT] post-merge updates

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (47 of 47 strings)

Translation: stock-logistics-reporting-17.0/stock-logistics-reporting-17.0-stock_quant_history
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-17-0/stock-logistics-reporting-17-0-stock_quant_history/it/

bosd committed:
Added translation using Weblate (Dutch)

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 100.0% (46 of 46 strings)

Translation: stock-logistics-reporting-17.0/stock-logistics-reporting-17.0-stock_quant_history
Translate-URL: https://translation.odoo-community.org/projects/stock-logistics-reporting-17-0/stock-logistics-reporting-17-0-stock_quant_history/nl/

bosd committed:
[MIG] stock_quant_history: Migration to 18.0

bosd committed:
[IMP] stock_quant_history: pre-commit execution

petrus-v commented:
> @petrus-v
> Shall we first migrate this base, then forward port all the improvements, from V14 and V17?
> As some of the "improvement" PR's are still open and have some review comments.

Both are fine to me as long there are mainly improvements !

petrus-v reviewed (approved):
Thanks to taking care to this migration

bosd committed:
[REF]stock_quant_history: Invert snapshot logic to use allowed locations

The previous implementation of the stock quant history snapshot used a
negative logic, ignoring a list of specified location usages. This was
causing issues with virtual production locations and made the module
hard to maintain.

This commit refactors the logic to be positive. Instead of ignoring
locations, the logic now only includes locations with the 'internal'
usage type. This simplifies the code, improves maintainability, and
resolves the issue with virtual locations.

The `_ignored_location_usage` method has been renamed to
`_allowed_location_usage` to better reflect its new purpose. The
corresponding unit tests have also been updated to align with the new
logic.

bosd committed:
[IMP]stock_quant_history: refactor domain filter to be more performant and support single and multiple locations

Dranyel-Bosd reviewed (approved):
LGTM :100:

bosd commented:
@JordiBForgeFlow Can you please merge?

bosd commented:
Please keep open!

bosd commented:
Ohh wait. it should be ready to merge!!

bosd commented:
@oca-maintainers/logistics-maintainers 
cc: @sebalix , @JordiBForgeFlow
I think this one fermented enough. Can you please merge :pray:

rousseldenis commented:
/ocabot migration stock_quant_history

rousseldenis reviewed (approved):
LGTM. Great, thanks for the improvements!

rousseldenis commented:
/ocabo merge nobump

rousseldenis commented:
/ocabot merge nobump
